> Just saw the CI failed. Is this related or just a flake?
> 
> ```
> FAILED examples/durable/test_durable.py::test_dag_spawn_returns_full_output - hatchet_sdk.exceptions.FailedTaskRunExceptionGroup: Workflow run 1584e75d-2775-4201-948e-68d0f9444247 failed.
> 
> --- Exception 1 ---
> KeyError: 'dag_child_1'
> Traceback (most recent call last):
>   File "/home/runner/work/hatchet/hatchet/sdks/python/hatchet_sdk/worker/runner/runner.py", line 269, in inner_callback
>     output = task.result()
>   File "/home/runner/work/hatchet/hatchet/sdks/python/hatchet_sdk/worker/runner/runner.py", line 361, in async_wrapped_action_func
>     return await task.aio_call(ctx, dependencies)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/runner/work/hatchet/hatchet/sdks/python/hatchet_sdk/runnables/task.py", line 462, in aio_call
>     return await self._fn(workflow_input, cast(Context, ctx), **dependencies)  # type: ignore
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/runner/work/hatchet/hatchet/sdks/python/examples/durable/worker.py", line 103, in durable_spawn_many_dags
>     child_index=result["dag_child_1"]["child_index"],
>                 ~~~~~~^^^^^^^^^^^^^^^
> KeyError: 'dag_child_
> ```

Was this on the `old_engine_new_sdk` test? If yes, then yeah this is expected to fail on the old engine, since we don't have the fix yet so it's reproducing the bug

--- a/pkg/repository/output.go
+++ b/pkg/repository/output.go
@@ -159,6 +159,10 @@ func ExtractOutputFromMatchData(data []byte) ([]byte, error) {
 		}
 
 		if len(keyMap) == 1 {
+			// this is a special case: if the child is a DAG, we want to return the output
+			// of each task in the DAG keyed by the task name (below), but if it's a standalone task,
+			// we want to return the task's output directly without the extra nesting, so we have to handle this
+			// situation separately
 			if entries, ok := keyMap["output"]; ok && len(entries) > 0 {
 				var event TaskOutputEvent
 				if err := json.Unmarshal(entries[0], &event); err != nil {

